File: backend/app/schemas/subscription.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class SubscriptionBase(BaseModel):
    plan: str
    status: str
    monthly_limit: int
    used_this_month: int = 0
    created_at: datetime | None = None


class Subscription(SubscriptionBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int

File: frontend/src/types/subscription.ts
export interface Subscription {
  id: number
  user_id: number
  plan: string
  status: string
  monthly_limit: number
  used_this_month: number
  created_at: string
}

export interface Plan {
  id: string
  name: string
  price: number
  features: string[]
  limit: number
}
